services/src/main/java/org/exoplatform/officeonline/WOPIService.java

```diff
@@ -487,14 +506,20 @@ public String renameFile(String fileId, EditorConfig config, String newTitle, St
   }
 
   /**
-   * Checks if current user can rename the document.
+   * Checks parent node permissions.
    *
    * @param node the node
    * @return true if user can rename
    * @throws RepositoryException the repository exception
    */
-  protected boolean canRenameDocument(Node node) throws RepositoryException {
-    NodeImpl parent = (NodeImpl) node.getParent();
+  protected boolean hasParentPermissions(Node node) throws RepositoryException {
```

Comment:
A better name would be `canModifyDocument()`

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/10/commits/36cfb2d370ee05b1aab1b09e471285054e5971e8